libs/service-portal/documents/src/components/DocumentCard/DocumentCard.tsx

```diff
@@ -33,46 +43,65 @@ interface Props {
 const DocumentCard: FC<Props> = ({ document }) => {
   const { formatMessage } = useLocale()
   const [isModalOpen, setIsModalOpen] = useState(false)
-  const { fetchDocument, loading, data, error } = useLazyDocumentDetail(
-    document.id,
-  )
-  useEffect(() => {
-    if (data || error) {
-      handleOnFetch()
-    }
-  }, [data, error])
+  const [{ loading, documentDetails }, setDocumentDetails] = useState<{
+    loading?: boolean
+    documentDetails?: DocumentDetails
+  }>({})
 
-  const handleOnFetch = () => {
-    if ((data?.fileType || '').toLowerCase() === 'pdf' && data?.content) {
-      downloadAsPdf(data.content)
-      return
-    }
-    if (data?.url) {
-      setIsModalOpen(true)
+  const displayErrorToast = () => {
+    toast.error(
+      formatMessage({
+        id: 'sp.documents:documentCard-error-singleDocument',
+        defaultMessage: 'Ekki tókst að sækja skjal',
+      }),
+    )
+  }
+
+  const displayDocument = (doc: DocumentDetails) => {
+    if (documentIsPdf(doc)) {
+      window.open(getPdfURL(doc.content))
       return
     }
 
-    if (error && !loading) {
-      toast.error(
-        formatMessage({
-          id: 'sp.documents:documentCard-error-singleDocument',
-          defaultMessage: 'Ekki tókst að sækja skjal',
-        }),
-      )
-    }
+    // Note: if document is not pdf, we need to log it into sentry and add it into the edge case.
+    setIsModalOpen(true)
   }
 
-  const handleOnClick = () => {
-    if (loading) {
-      return
-    }
-    if (!data) {
-      fetchDocument()
-    } else {
-      handleOnFetch()
+  const fetchDocument = async () => {
+    setDocumentDetails({ loading: true })
+
+    // Note: opening window before fetching data, to prevent popup-blocker
+    let windowRef = window.open()
+    try {
+      const { data } = await client.query({
+        query: GET_DOCUMENT,
+        variables: { input: { id: document.id } },
+      })
+      const doc = data?.getDocument
+      if (!doc) {
+        throw new Error()
```

Comment:
Should probably have some kind of description of which error is being thrown here.